client/packages/plugins/README.md

```diff
@@ -1,12 +1,310 @@
+- creating a plugin
+- using standard app components
+- use of the app theme
+- the export of two plugin components
+- fetching data using the graphQL API
+- utilising utility functions from the app
+
+**Stock Donor**
+This example adds a new field to a stock line, displaying the stored data in a new column within the list view and allowing editing of the field in the detail view. It demonstrates:
+- creating a plugin
+- using standard app components
+- use of the app theme
+- implementing a column plugin
+- fetching data using the graphQL API
+- inserting and updating data using graphQL
+- utilising utility functions from the app
+- using plugin events to 
+  - trigger validation in the host page
+  - save data when the host page is saving
+- store data which is specific to the plugin
+
+Create a new directory under the `client/packages/plugins` directory. In this you will need:
+
+- src : containing the source files of the plugin
+- plugin.json : which defines the plugin
+- webpack.config.js : used for bundling the plugin
+- dist : build output, not committed to source control
+
+## Development process 
+
+Create a react component and update webpack.config.js, adding the component files to the `exposes` section:
+
+```
+  plugins: [
+    new ModuleFederationPlugin({
+      name: 'StockDonor',
+      remotes: {},
+      exposes: {
+        StockDonorEdit: './src/StockDonorEdit.tsx',
+        StockDonorEditInput: './src/StockDonorEditInput.tsx',
+        StockDonorColumn: './src/StockDonorColumn.tsx',
+      },
+      shared: {
+        ...dependencies,
+        react: {
+          eager: true,
+          singleton: true,
+          requiredVersion: dependencies['react'],
+        },
+        'react-dom': {
+          eager: true,
+          singleton: true,
+          requiredVersion: dependencies['react-dom'],
+        },
+        'react-singleton-context': { singleton: true, eager: true },
+      },
+    }),
+```
+
+In development mode simply run as usual (`yarn start` or `yarn start-local`) and webpack will read any plugin files from the `client/packages/plugins` directly and include them in the bundled javascript.
+
+When you are ready, run `yarn build-plugins` from the `./client` directory. This will compile and bundle all of the plugins. The contents of the `./client/packages/plugins/[your plugin name]/dist` folder can then be copied to the `./server/app_data/plugins/[your plugin name]` - available to be read by the server.
+
+The server will read all folders under `./server/app_data/plugins` and for any folders that have a `plugin.json` file, will make that plugin available in the list.
+
+### Things to note
+
+When plugins are running in 'production' mode, the standard react contexts are not available. The package `react-singleton-context` is used instead of the standard react context in order to share the context across the two app environments in this case. It requires the use of proxy providers for example:
```

Comment:
```suggestion
When plugins are running in 'production' mode, the standard react contexts are not available. The package `react-singleton-context` is used instead of the standard react context in order to share the context across the two app environments. It requires the use of proxy providers:
```